Kernel/SharedInterruptHandler.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <AK/Types.h>
+#include <AK/NonnullOwnPtr.h>
+#include <AK/HashTable.h>
+
+class InterruptHandler;
+class SharedInterruptHandler final {
+public:
+
+    static void initialize(u8 interrupt_number);
+    static SharedInterruptHandler& get_shared_interrupt_handler(u8 interrupt_number);
+    ~SharedInterruptHandler();
+    void handle_interrupt();
+
+    u8 interrupt_number() const { return m_interrupt_number; }
+
+    void register_interrupt_handler(InterruptHandler&);
+    void unregister_interrupt_handler(InterruptHandler&);
+
+protected:
+    void enable_interrupt_vector();
+    void disable_interrupt_vector();
+    explicit SharedInterruptHandler(u8 irq);
+
+private:
+    HashTable<InterruptHandler*> m_handlers;
+    u8 m_interrupt_number { 0 };
```

Comment:
It would be good to be consistent about whether we call these "IRQ" or "interrupt number".

Reply:
That's a good idea, and therefore I made some changes to replicate what's going on.
Let's remember that currently we use IRQs, but in the future (hopefully in the near future), we'll use MSIs also (especially for PCIe devices that mush support MSIs). So now we are handling IRQs only but that won't be true always, and we should give names that consider this plan.